src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -528,6 +533,47 @@ GetRenameShardIndexCommand(char *indexName, uint64 shardId)
 }
 
 
+/*
+ * RenameShardRelationStatistics appends given shardId to the end of the names
+ * of shard relation statistics. This function utilizes GetExplicitStatsNameList
+ * to pick the statistics to be renamed, see more details in function's comment.
+ */
+static void
+RenameShardRelationStatistics(Oid shardRelationId, uint64 shardId)
+{
+	List *statsOidList = GetExplicitStatisticsIdList(shardRelationId);
+	List *statsNameList = GetExplicitStatsNameList(statsOidList);
+
+	char *statName = NULL;
+	foreach_ptr(statName, statsNameList)
+	{
+		const char *commandString = GetRenameShardStatsCommand(statName, shardId);
+		ExecuteAndLogDDLCommand(commandString);
+	}
+}
+
+
+/*
+ * GetRenameShardStatsCommand returns DDL command to append given shardId to
+ * the statistics with statName.
+ */
+static char *
+GetRenameShardStatsCommand(char *statName, uint64 shardId)
+{
+	char *shardStatsName = pstrdup(statName);
+	AppendShardIdToName(&shardStatsName, shardId);
+	const char *quotedShardStatsName = quote_identifier(shardStatsName);
+
+	const char *quotedStatsName = quote_identifier(statName);
```

Comment:
should we qualify schema name ? (see other such functions in `create_citus_local_table.c`)

Reply:
(maybe not needed if push/pop current search path when finding those stat objects)